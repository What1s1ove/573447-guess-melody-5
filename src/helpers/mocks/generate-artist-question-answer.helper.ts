import { IArtistQuestionAnswer } from '~/common/interfaces/interfaces';
import { ArtistName } from '~/common/enums/enums';

const AVATAR_URL = `https://api.adorable.io/avatars/128`;

const generateArtistQuestionAnswer = (
  artist: ArtistName
): IArtistQuestionAnswer => ({
  artist,
  picture: `${AVATAR_URL}/${Math.random()}`,
});

export { generateArtistQuestionAnswer };
