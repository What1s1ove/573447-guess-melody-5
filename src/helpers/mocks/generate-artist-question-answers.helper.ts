import { generateArtistQuestionAnswer } from '~/helpers/mocks/generate-artist-question-answer.helper';
import { IArtistQuestionAnswer } from '~/common/interfaces/interfaces';

const generateArtistQuestionAnswers = (
  artists: string[]
): IArtistQuestionAnswer[] => {
  const artistQuestionAnswers = artists.map((artist) =>
    generateArtistQuestionAnswer(artist)
  );

  return artistQuestionAnswers;
};

export { generateArtistQuestionAnswers };
