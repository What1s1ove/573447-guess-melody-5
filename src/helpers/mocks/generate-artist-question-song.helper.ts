import { ArtistName } from '~/common/enums/enums';
import { IArtistQuestionSong } from '~/common/interfaces/game/game';

const generateArtistQuestionSong = (
  artist: ArtistName
): IArtistQuestionSong => ({
  artist,
  src: `https://upload.wikimedia.org/wikipedia/commons/4/4e/BWV_543-fugue.ogg`,
});

export { generateArtistQuestionSong };
