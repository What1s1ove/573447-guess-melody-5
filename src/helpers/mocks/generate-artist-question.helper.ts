import { getRandomItem, getShuffledArray } from '~/helpers/array/array';
import { generateArtistQuestionSong } from '~/helpers/mocks/generate-artist-question-song.helper';
import { generateArtistQuestionAnswer } from '~/helpers/mocks/generate-artist-question-answer.helper';
import { generateArtistQuestionAnswers } from '~/helpers/mocks/generate-artist-question-answers.helper';
import { getUniqueAnswerTypes } from '~/helpers/mocks/get-unique-answer-types.helper';
import { IArtistQuestion } from '~/common/interfaces/interfaces';
import { ArtistName, QuestionType } from '~/common/enums/enums';

const CUT_GENRE_START_IDX = 0;
const ANSWERS_COUNT = 2;

const artists = Object.values(ArtistName);

const generateArtistQuestion = (): IArtistQuestion => {
  const artist = getRandomItem(artists);
  const song = generateArtistQuestionSong(artist);
  const answer = generateArtistQuestionAnswer(artist);
  const uniqueArtists = getUniqueAnswerTypes(artist, artists);
  const answers = generateArtistQuestionAnswers(
    getShuffledArray(uniqueArtists.slice(CUT_GENRE_START_IDX, ANSWERS_COUNT))
  );

  return {
    song,
    type: QuestionType.ARTIST,
    answers: [answer, ...answers],
  };
};

export { generateArtistQuestion };
