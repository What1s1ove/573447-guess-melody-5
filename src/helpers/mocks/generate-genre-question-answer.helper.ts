import { GenreType } from '~/common/enums/enums';
import { IGenreQuestionAnswer } from '~/common/interfaces/interfaces';

const generateGenreQuestionAnswer = (
  genre: GenreType
): IGenreQuestionAnswer => ({
  genre,
  src: `https://upload.wikimedia.org/wikipedia/commons/4/4e/BWV_543-fugue.ogg`,
});

export { generateGenreQuestionAnswer };
