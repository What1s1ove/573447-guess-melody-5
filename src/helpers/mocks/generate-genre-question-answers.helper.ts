import { generateGenreQuestionAnswer } from '~/helpers/mocks/generate-genre-question-answer.helper';
import { GenreType } from '~/common/enums/enums';
import { IGenreQuestionAnswer } from '~/common/interfaces/interfaces';

const generateGenreQuestionAnswers = (
  genres: GenreType[]
): IGenreQuestionAnswer[] => {
  const genreQuestionAnswers = genres.map((genre) => {
    const genreQuestionAnswer = generateGenreQuestionAnswer(genre);

    return genreQuestionAnswer;
  });

  return genreQuestionAnswers;
};

export { generateGenreQuestionAnswers };
