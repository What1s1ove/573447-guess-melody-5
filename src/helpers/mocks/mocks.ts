export * from './generate-questions.helper';
export * from './generate-question.helper';
export * from './generate-genre-question-answer.helper';
export * from './generate-genre-question-answers.helper';
export * from './get-unique-genre-types.helper';
export * from './generate-genre-question.helper';
export * from './generate-artist-question-song.helper';
