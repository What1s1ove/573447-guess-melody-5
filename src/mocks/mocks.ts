export * from './game/games';
export * from './store/store';
