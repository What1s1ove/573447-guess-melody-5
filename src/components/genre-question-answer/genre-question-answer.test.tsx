import * as React from 'react';
import { render } from 'enzyme';
import { mockedGenreQuestionAnswer } from '~/mocks/mocks';
import GenreQuestionAnswer from './genre-question-answer';

describe(`GenreQuestionAnswer component`, () => {
  it(`should render GenreQuestionAnswer component`, () => {
    const component = render(
      <GenreQuestionAnswer
        isChecked
        answer={mockedGenreQuestionAnswer}
        idx={1}
        renderPlayer={jest.fn()}
        onAnswerChange={jest.fn()}
      />
    );

    expect(component).toMatchSnapshot();
  });
});
