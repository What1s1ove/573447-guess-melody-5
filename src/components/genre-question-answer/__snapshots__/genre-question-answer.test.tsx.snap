// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GenreQuestionAnswer component should render GenreQuestionAnswer component 1`] = `
<div
  class="track"
>
  <div
    class="game__answer"
  >
    <input
      checked=""
      class="game__input visually-hidden"
      id="answer-1"
      name="answer"
      type="checkbox"
      value="1"
    />
    <label
      class="game__check"
      for="answer-1"
    >
      Отметить
    </label>
  </div>
</div>
`;
